### PR TITLE
Clean HealthPublisher base from SF dependency

### PR DESCRIPTION
**Pull request by mvxxx**
## Description
Changed little implementation of HealthCheckPublisher base in order to avoid SF dependency in Health check publisher base

## Tests
Unit tests pass, no need for new ones. Tested on sample service.

### CODE REVIEW

File: src/Extensions/Diagnostics.HealthChecks/Internal/HealthCheckPublisher.cs [hunk 1]
```diff
@@ -3,55 +3,41 @@
...
 using System.Threading.Tasks;
 using Microsoft.Extensions.Diagnostics.HealthChecks;
 using Microsoft.Extensions.ObjectPool;
-using ServiceFabricHealth = System.Fabric.Health;
 
 namespace Microsoft.Omex.Extensions.Diagnostics.HealthChecks
 {
 	internal abstract class HealthCheckPublisher : IHealthCheckPublisher
 	{
-		protected readonly ObjectPool<StringBuilder> StringBuilderPool;
```
Comment (AndreyTretyak) [inline src/Extensions/Diagnostics.HealthChecks/Internal/HealthCheckPublisher.cs:18]: Why are we changing visibility for this field?

File: src/Extensions/Diagnostics.HealthChecks/Internal/HealthCheckPublisher.cs [hunk 2]
```diff
@@ -3,55 +3,41 @@
...
 
-		public abstract Task PublishAsync(HealthReport report, CancellationToken cancellationToken);
+		public abstract Task PublishAsync(HealthReport report, CancellationToken cancellationToken);
 
-		private ServiceFabricHealth.HealthInformation FinalizeHealthReport(string healthReportSummaryProperty, ServiceFabricHealth.HealthState healthState)
-		{
-			return new ServiceFabricHealth.HealthInformation(HealthReportSourceId, healthReportSummaryProperty, healthState);
-		}
-		protected void PublishAllEntries(HealthReport report, Action<ServiceFabricHealth.HealthInformation> publishFunc,
+		protected void PublishAllEntries(HealthReport report, Func<string, HealthStatus, string, Task> publishFunc,
```
Comment (AndreyTretyak) [inline src/Extensions/Diagnostics.HealthChecks/Internal/HealthCheckPublisher.cs:26]: I'm not sure that using Func here is a good approach, parameter list like this `string, HealthStatus, string, Task` very confusing. Could we make it an abstract function? Or extract functionality into the interface.

**Review by AndreyTretyak — DISMISSED** [review]

File: src/Extensions/Diagnostics.HealthChecks/Internal/HealthCheckPublisher.cs [hunk 3]
```diff
@@ -3,55 +3,40 @@
...
 		internal abstract string HealthReportSourceId { get; }
 
-		internal const string HealthReportSummaryProperty = "HealthReportSummary";
+		internal const string HealthReportSummaryProperty = "HealthReportSummary";
 
-		public HealthCheckPublisher(ObjectPoolProvider objectPoolProvider) => StringBuilderPool = objectPoolProvider.CreateStringBuilderPool();
+		public HealthCheckPublisher(ObjectPoolProvider objectPoolProvider) => StringBuilderPool = objectPoolProvider.CreateStringBuilderPool();
 
-		public abstract Task PublishAsync(HealthReport report, CancellationToken cancellationToken);
+		public abstract Task PublishAsync(HealthReport report, CancellationToken cancellationToken);
```
Comment (AndreyTretyak) [inline src/Extensions/Diagnostics.HealthChecks/Internal/HealthCheckPublisher.cs:24]: nit: trailing whitespace at the end

**Review by AndreyTretyak — DISMISSED** [review]

File: src/Extensions/Diagnostics.HealthChecks/Internal/ServiceFabricHealthCheckPublisher.cs [hunk 4]
```diff
@@ -38,19 +40,49 @@ public override Task PublishAsync(HealthReport report, CancellationToken cancell
...
+			{
+				// Ignore, the service instance is closing.
+			}
 
 			return Task.CompletedTask;
+		}
+
+		protected override void PublishHealthReportEntry(string healthCheckName, HealthStatus status, string description)
+		{
+			if(m_reportHealth == null)
```
Comment (dimeehan) [inline src/Extensions/Diagnostics.HealthChecks/Internal/ServiceFabricHealthCheckPublisher.cs:67]: space missing after if

File: src/Extensions/Diagnostics.HealthChecks/Internal/ServiceFabricHealthCheckPublisher.cs [hunk 5]
```diff
@@ -38,19 +40,49 @@ public override Task PublishAsync(HealthReport report, CancellationToken cancell
...
+			{
+				m_logger.LogWarning(Tag.Create(), "Publisher run before health report functions is provided.");
+				return;
+			}
+
+			ServiceFabricHealth.HealthInformation healthEntry = new(HealthReportSourceId, healthCheckName, ToSfHealthState(status));
+			healthEntry.Description = description;
+			m_reportHealth(healthEntry);
 		}
+
```
Comment (dimeehan) [inline src/Extensions/Diagnostics.HealthChecks/Internal/ServiceFabricHealthCheckPublisher.cs:77]: nit: remove extra line

**Review by dimeehan — APPROVED** [review]

**Review by dimeehan — APPROVED** [review]

**Review by AndreyTretyak — APPROVED** [review]

**Pull request merged**